detox/src/DetoxWorker.js

```diff
@@ -97,7 +97,7 @@ class DetoxWorker {
     this._deviceConfig = deviceConfig;
     this._sessionConfig = sessionConfig;
     // @ts-ignore
-    this._sessionConfig.sessionId = sessionConfig.sessionId || uuid.UUID();
```

Comment:
When multiple test suites are present in a session, web socket connection is closed after the execution of each test suite, and a new web socket connection is created, thus web socket id changes, but session id remains constant for each test suite. Therefore, to ensure that the same device is used for all the test suites in a test, we want this unique identifier(session_id). Thus, we cannot let users define this.